pallets/nfts/src/lib.rs

```diff
@@ -693,6 +736,24 @@ pub mod pallet {
 		CollectionNotEmpty,
 		/// The witness data should be provided.
 		WitnessRequired,
+		/// The account owns zero items in the collection.
+		NoItemOwned,
+		/// The collection has existing approvals.
+		CollectionApprovalsExist,
+		/// Collection and item approval conflicts.
+		DelegateApprovalConflict,
+	}
+
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I> {
+		#[cfg(any(feature = "std", test))]
+		fn integrity_test() {
+			use core::any::TypeId;
+			assert!(
+				TypeId::of::<<T as Config<I>>::ItemId>() != TypeId::of::<u64>() &&
```

Comment:
Note: Context for this integration test is to make sure `ItemId` won't exceed the `u32` type. If so, the overflow case can happen because the type used in `AccountBalance` is u32.

Reply:
Depends on the response in https://github.com/r0gue-io/pop-node/pull/387/files/5615f1b622de0c9675303a206d70f39defeb18a9#r1859163342, this can be removed.